src/components/Dialogs/BoardDialog.vue

```diff
@@ -101,22 +103,24 @@ const formData = ref<{ name: string; columns: string[] }>({
 })
 const formNameError = ref(false)
 
-const submit = async () => {
-  if (formData.value.name === '') return
-
-  emits('close-dialog')
-
-  if (props.action === 'add') {
-    await addNewBoard({
+const submitFunctions = {
+  add: () =>
+    addNewBoard({
+      userID: userID.value,
+      name: formData.value.name,
+      columns: formData.value.columns.map((name) => ({ name, tasks: [] }))
+    }),
```

Comment:
It is used here:
```js
// boardsNew.ts
const getBoardsData = (userID: Board['userID']) => {
    if (userID) {
      userBoardsColRef.value = query(
        boardsColRef,
        where('userID', '==', userID),
        orderBy('createdAt', 'desc')
      )
      // ...
   }
   ```
   The thing is - in my opinion - the scalability: notice I solely have the `boards` collection. If I had like a bunch of users that have let's say a million boards in total, I can't fetch them all because my app would never load. And even if I could, I'd still need this user ID for me to display only their boards to a user.

Obviously, this `query` thing doesn't solve this issue completeley, because firebase still needs to loop through all those boards, which I believe still impacts the app performance. That's why I think my previous approach with the `users` collection containing all their boards was more performant but since it couldn't be actually applied... it is what is is.